aspnetcore/blazor/hybrid/class-libraries.md

```diff
@@ -0,0 +1,483 @@
+
+* [.NET](https://dotnet.microsoft.com/download/dotnet)
+* [ASP.NET Core](https://dotnet.microsoft.com/apps/aspnet)
+* [Blazor](https://dotnet.microsoft.com/apps/aspnet/web-apps/blazor)
+* [.NET MAUI](https://dotnet.microsoft.com/apps/maui)
+* [Azure Container Apps](https://azure.microsoft.com/services/container-apps/)
+* [Orleans](/dotnet/orleans/overview)
+
+## Share web UI Razor components, code, and static assets
+
+Components from an RCL can be simultaneously shared by web and native client apps built using Blazor. The guidance in <xref:blazor/components/class-libraries> explains how to share Razor components using a Razor class library (RCL). The same guidance applies to reusing Razor components from an RCL in a Blazor Hybrid app.
+
+Component namespaces are derived from the RCL's package ID or assembly name and the component's folder path within the RCL. For more information, see <xref:blazor/components/index#namespaces>. [`@using`](xref:mvc/views/razor#using) directives can be placed in `_Imports.razor` files for components and code, as the following example deomonstrates for an RCL named `SharedLibrary` with a `Shared` folder of shared Razor components and a `Data` folder of shared data classes:
+
+```razor
+@using SharedLibrary
+@using SharedLibrary.Shared
+@using SharedLibrary.Data
+```
+
+Place shared static assets in the RCL's `wwwroot` folder and update `<link>` element paths:
+
+> :::no-loc text="_content/{PACKAGE ID/ASSEMBLY}/{PATH}/{FILENAME}":::
+
+Placeholders:
+
+* `{PACKAGE ID/ASSEMBLY}`: The package ID or assembly name of the RCL.
+* `{PATH}`: Optional path within the RCL's `wwwroot` folder.
+* `{FILENAME}`: The filename of the static asset.
+
+For an RCL named `SharedLibrary` and using the minified Bootstrap stylesheet as an example:
+
+> :::no-loc text="_content/SharedLibrary/css/bootstrap/bootstrap.min.css":::
+
+For additional information on how to share static assets across projects, see the following articles:
+
+* <xref:blazor/components/class-libraries>
+* <xref:razor-pages/ui-class>
+
+The root `index.html` file is specific to the app and should remain in the Blazor Hybrid app.
+
+The root Razor Component (`App.razor` or `Main.razor`) can be shared, but often might need to be specific to the hosting app. For example, `App.razor` is different in the Blazor Server and Blazor WebAssembly project templates when authentication is enabled for apps. You can add the [`AdditionalAssemblies` parameter](xref:blazor/fundamentals/routing#route-to-components-from-multiple-assemblies) to specify the location of any shared routable components, and you can specify a [shared default layout component for the router](xref:blazor/fundamentals/routing#route-templates) by type name.
+
+## Provide code and services independent of hosting model
+
+When code must differ for across hosting models, abstract the code as interfaces and inject the service implementations into each project.
+
+### Example 1
```

Comment:
This first example shows how to deal with multiple target platforms. I think it might be simpler to start with the weather forecast service example, which shows how to deal with different hosting models, and then show the more complicated example of dealing with multiple target platforms.

I believe the Platforms folder convention is also specific to .NET MAUI class libraries that have UseMaui set to true in the project. I don't think we support setting UseMaui to true in an RCL. So you'd need to move the platform specific service implementations into a separate .NET MAUI class library.

This still leaves the question of how to package and distribute the Blazor components with the platform specific implementations. I'm not sure what we should recommend for that. @Eilon @javiercn Thoughts?

Reply:
> start with the weather forecast service example

**_Done!_** :+1:

The rest of your remarks are cryptic to me, so I'll wait for more feedback on the exact actions to take.

Reply:
I talked this over with @SteveSandersonMS and we agreed to hold off on diving into the complexity of dealing with platform specific logic for now. At this point I think we should remove the 2nd example and just have the first weather forecast example.I would also like to change the weather forecast example to recommend that the `IWeatherForecastService` implementations go into the corresponding Blazor apps instead of in the RCL.

Reply:
**_Updates made._** ... I held the 2nd example in place in case it's used later (or it's the basis of something used later).